sklearn/decomposition/kernel_pca.py

```diff
@@ -253,8 +273,10 @@ def fit(self, X, y=None):
         self._fit_transform(K)
 
         if self.fit_inverse_transform:
-            sqrt_lambdas = np.diag(np.sqrt(self.lambdas_))
-            X_transformed = np.dot(self.alphas_, sqrt_lambdas)
+            # Transform X
+            # (shortcut since we transform the same X that was used to fit)
+            X_transformed = self.alphas_ * np.sqrt(self.lambdas_)
+
```

Comment:
This is not related to the issue, right?

If that's the case this would be best in another PR if you think it's worth it

Reply:
It is related, actually. In order for the code to stay readable/maintainable (so that such a bug does not appear again), I aligned the codes that do exactly the same thing, so that they are identical to the reader. 

In the case you mention, [the two lines in `fit()`](https://github.com/scikit-learn/scikit-learn/pull/12143/files#diff-d6cfec34cb25a8568bc721be1ce40e44L256) were doing exactly the same thing than [the single line in `fit_transform()`](https://github.com/scikit-learn/scikit-learn/pull/12143/files#diff-d6cfec34cb25a8568bc721be1ce40e44L278), so I changed the first so that they are identical to the second.

Reply:
Right that makes sense